### PR TITLE
Add define:webserver hook

### PR DESCRIPTION
**Pull request by usergenic**
- Allows a callback to reassign the express application to provide mechanism to front polyserve with alternate request handlers and middleware.
 - [x] CHANGELOG.md has been updated

### CODE REVIEW

**Comment by usergenic** [comment]
cc @thedeeno @bubbit wrt providing a hook to support plugins such as wct-istanbul which need to get in front of the default polyserve request handlers.

File: runner/context.ts [hunk 1]
```diff
@@ -110,7 +113,11 @@ export class Context extends events.EventEmitter {
    * @return {!Context}
    */
   emitHook(
-      name: 'prepare:webserver', app: Express.Application,
+      name: 'define:webserver', app: express.Express, options: ServerOptions,
+      mapper: (app: Express.Application) => void,
```
Comment (justinfagnani) [inline runner/context.ts]: document the new parameters

File: runner/webserver.ts [hunk 2]
```diff
@@ -167,15 +167,26 @@ Expected to find a ${mdFilenames.join(' or ')} at: ${pathToLocalWct}/
...
+        {
+          root: options.root,
+          compile: options.compile,
+          hostname: options.webserver.hostname,
+          headers: DEFAULT_HEADERS,
+          packageName,
+          additionalRoutes,
+          npm: !!options.npm,
+        },
+        async (app: express.Express, options: ServerOptions) => {
```
Comment (justinfagnani) [inline runner/webserver.ts]: can you pull this out into a named variable?

File: runner/webserver.ts [hunk 3]
```diff
@@ -167,15 +167,26 @@ Expected to find a ${mdFilenames.join(' or ')} at: ${pathToLocalWct}/
...
+          await wct.emitHook(
```
Comment (justinfagnani) [inline runner/webserver.ts]: comment the purpose of the hook for clarity for readers

**Pull request merged**